tests/pass/tree-borrows/transmute-unsafecell.rs

```diff
@@ -1,13 +1,32 @@
 //@compile-flags: -Zmiri-tree-borrows
 
+//! Testing `mem::transmute` between types with and without interior mutability.
+//! All transmutations should work, as long as we don't do any actual accesses
+//! that violate immutability.
+
 use core::cell::UnsafeCell;
 use core::mem;
 
 fn main() {
     unsafe {
-        let x = &0i32;
-        // As long as we only read, transmuting this to UnsafeCell should be fine.
-        let cell_x: &UnsafeCell<i32> = mem::transmute(&x);
-        let _val = *cell_x.get();
+        ref_to_cell();
+        cell_to_ref();
     }
 }
+
+// Pretend that the reference has interior mutability.
+// Don't actually mutate it though, it will fail because it has a Frozen parent.
```

Comment:
do we have tests for that? Similarly for the other direction, mutating the unsafe cell should invalidate reading from the direct reference